### ISSUE TITLE
[BUG] Update Xamarin Forms Samples to work with latest Xamarin

### ISSUE THREAD

**Issue by parthchopra** [reactiveui/ReactiveUI.Samples#288]
### Update Xamarin Forms Samples to work with latest Xamarin

**Build Errors**
```
Project "/Users/pchopra/Projects/ReactiveUI.Samples/xamarin-forms/MasterDetail/MasterDetail/MasterDetail.csproj" (Build target(s)):

Target WinUICheckTargetPlatformVersion:
    /Users/pchopra/.nuget/packages/xamarin.forms/4.3.0.991250/build/Xamarin.Forms.targets(189,5): warning : Xamarin.Forms recommends TargetPlatformMinVersion >= 10.0.14393.0 (current project is -1)
    /Users/pchopra/.nuget/packages/xamarin.forms/4.3.0.991250/build/Xamarin.Forms.targets(195,5): warning : Xamarin.Forms recommends TargetPlatformVersion >= 10.0.17763.0 (current project is -1)
Done building target "WinUICheckTargetPlatformVersion" in project "MasterDetail.csproj".
Target ResolvePackageAssets:
    /Users/pchopra/Projects/ReactiveUI.Samples/xamarin-forms/MasterDetail/MasterDetail/MasterDetail.csproj : error NU1605: Detected package downgrade: Xamarin.Forms from 4.7.0.1351 to 4.3.0.991250. Reference the package directly from the project to select a different version. 
    /Users/pchopra/Projects/ReactiveUI.Samples/xamarin-forms/MasterDetail/MasterDetail/MasterDetail.csproj : error NU1605:  MasterDetail -> ReactiveUI.XamForms 12.1.5 -> Xamarin.Forms (>= 4.7.0.1351) 
    /Users/pchopra/Projects/ReactiveUI.Samples/xamarin-forms/MasterDetail/MasterDetail/MasterDetail.csproj : error NU1605:  MasterDetail -> Xamarin.Forms (>= 4.3.0)
Done building target "ResolvePackageAssets" in project "MasterDetail.csproj" -- FAILED.

Done building project "MasterDetail.csproj" -- FAILED.

Build FAILED.

/Users/pchopra/.nuget/packages/xamarin.forms/4.3.0.991250/build/Xamarin.Forms.targets(189,5): warning : Xamarin.Forms recommends TargetPlatformMinVersion >= 10.0.14393.0 (current project is -1)
/Users/pchopra/.nuget/packages/xamarin.forms/4.3.0.991250/build/Xamarin.Forms.targets(195,5): warning : Xamarin.Forms recommends TargetPlatformVersion >= 10.0.17763.0 (current project is -1)
/Users/pchopra/Projects/ReactiveUI.Samples/xamarin-forms/MasterDetail/MasterDetail/MasterDetail.csproj : error NU1605: Detected package downgrade: Xamarin.Forms from 4.7.0.1351 to 4.3.0.991250. Reference the package directly from the project to select a different version. 
/Users/pchopra/Projects/ReactiveUI.Samples/xamarin-forms/MasterDetail/MasterDetail/MasterDetail.csproj : error NU1605:  MasterDetail -> ReactiveUI.XamForms 12.1.5 -> Xamarin.Forms (>= 4.7.0.1351) 
/Users/pchopra/Projects/ReactiveUI.Samples/xamarin-forms/MasterDetail/MasterDetail/MasterDetail.csproj : error NU1605:  MasterDetail -> Xamarin.Forms (>= 4.3.0)
    2 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.09

========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

Cloned the repo and tried to build main branch but fails with the below errors

![image](https://user-images.githubusercontent.com/8428794/105104566-82b2c600-5a80-11eb-9d3d-66d3c840ec34.png)

**Environment**
VS Mac version: 8.8.5 (build 18)
Mono 6.12.0.113
Xamarin.iOS Version: 14.8.0.3
Xamarin.Android Version: 11.1.0.26

**Comment by parthchopra** [comment]
I can update the solutions and create a PR

**Comment by glennawatson** [comment]
PR will be appreciated thanks

**Comment by glennawatson** [comment]
Closing this PR, fixed by #289
